File: src/main/scala/gameover/fwk/ai/impl/ManhattanHComputeStrategy.scala
package gameover.fwk.ai.impl

import gameover.fwk.ai.{CollisionDetector, HComputeStrategy}

class ManhattanHComputeStrategy extends HComputeStrategy{
  override def h(x: Int, y: Int, px: Int, py: Int, collisionDetector: CollisionDetector) : Int = Math.abs(px - x) * 10 + Math.abs(py - y) * 10
}

File: src/main/kotlin/gameover/fwk/ai/impl/ManhattanHComputeStrategy.kt
package gameover.fwk.ai.impl

import gameover.fwk.ai.CollisionDetector
import gameover.fwk.ai.HComputeStrategy

class ManhattanHComputeStrategy : HComputeStrategy {
    override fun h(x: Int, y: Int, px: Int, py: Int, collisionDetector: CollisionDetector) : Int = Math.abs(px - x) * 10 + Math.abs(py - y) * 10
}